packages/users-app/src/screens/helper.ts

```diff
@@ -0,0 +1,55 @@
+import { SelectChangeEvent } from "@mui/material"
+import { User } from '../types'
+import { userSchema } from "../utils"
+
+export function handleOnChange(nestedKey: string, user: Readonly<User>, e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement> | SelectChangeEvent<string[]>, setUser: React.Dispatch<React.SetStateAction<Readonly<User>>>, validateUserField: (field: string, newBasicInfoState?: object) => void) {
+    let newBasicUserState
+  
+    if (nestedKey) {
+      newBasicUserState = {
+        ...user
+      }
```

Comment:
```suggestion
      newBasicUserState = structuredClone(user)
```

Safer deep cloning. It's helped me resolve quite a lot of bugs.